rllab/misc/tensor_summary.py

```diff
@@ -0,0 +1,89 @@
+import os
+
+import numpy as np
+import tensorflow as tf
+from tensorboard import summary as summary_lib
+from tensorboard.plugins.custom_scalar import layout_pb2
+
+from rllab.misc.console import mkdir_p
+
+
+class Summary:
+    def __init__(self):
+        self._summary_scale = tf.Summary()
+        self._scope_tensor = {}
+
+        self._session = tf.Session()
+
+        self._default_step = 0
+        self._step_key = None
+        self._writer = None
+
+    def set_dir(self, dir_name):
+        if not dir_name:
+            if self._writer:
+                self._writer.close()
+                self._writer = None
+        else:
+            mkdir_p(os.path.dirname(dir_name))
+            self._writer = tf.summary.FileWriter(dir_name)
+            self._default_step = 0
+            assert self._writer is not None
+            print("tensorboard data will be logged into:", dir_name)
+
+    def dump_tensorboard(self, step=None):
+        if not self._writer:
+            return
+        run_step = self._default_step
+        if step:
+            run_step = step
+        else:
+            self._default_step += 1
+
+        self.dump_scale(run_step)
+        self.dump_tensor(run_step)
+        self.layout_tensor()
+
+    def dump_graph(self):
+        self._writer.add_graph(tf.get_default_graph())
+        self._writer.flush()
+
+    def record_scale(self, key, val):
+        self._summary_scale.value.add(tag=str(key), simple_value=float(val))
+
+    def dump_scale(self, step):
+        self._writer.add_summary(self._summary_scale, int(step))
+        self._writer.flush()
+        del self._summary_scale.value[:]
+
+    def record_tensor(self, name, val):
+        scope = str(name).split('/', 1)[0]
+        if scope not in self._scope_tensor.keys():
+            self._scope_tensor[scope] = [name]
+        else:
+            if name not in self._scope_tensor[scope]:
+                self._scope_tensor[scope].append(name)
+            val = np.array(val).reshape(-1)
```

Comment:
shouldn't this be outside the `else` to update the value of `scope/name` in all cases?